### DISCUSSION TITLE
A couple ideas for enhancing GPS matching

### DISCUSSION THREAD

**Discussion by m-qin** [NSAPH-Software/CausalGPS#181] [Ideas]
Future features that might be nice to have for GPS matching, based on what the _MatchIt_ package can do for binary-treatment matching (https://kosukeimai.github.io/MatchIt/articles/matching-methods.html):

- exact matching (particularly for categorical, i.e., factor, covariates, allow the user to specify that observations can only be matched to other observations that share their exact value of this covariate)
- k:1 matching (allow user to specify that observations can only be matched to ≤k other observations)

**Comment by Naeemkh** [comment]
Thanks, @m-qin for the comments. It is an exciting idea; however, the implementation will be different than matchit. We do not match for the data sample. Instead, we match the data to the predefined exposure level. We can add a feature to select the next better match (under some conditions) in case of trying to match to the same data sample. I am moving this issue to the discussions.
